test/specs/errors/tests/07.http_409_error.js

```diff
@@ -6,51 +6,143 @@ exports.execute = function (options) {
+        });
+
+        it("if it's an integrity error, we should generate a more readable message.", function (done) {
+          reference1.delete().then(null, function (err) {
+              expect(err.message).toBe(integrityErrorMappedMessage, "invalid error message");
+              done();
+          }).catch(function(err) {
+              console.log(err);
+              done.fail();
+          });
         });
 
+
+
+        it("if it's an integrity error and no displayName was found then use table name passed by ermrest", function (done) {
+          reference2.delete().then(null, function (err) {
+              expect(err.message).toBe(integrityErrorMappedMessageWithoutDisplay, "invalid error message");
+              done();
+          }).catch(function(err) {
+              console.log(err);
+              done.fail();
+          });
+        });
+
+        it("if it's an integrity error and pure and binary association then to_name should be displayed", function (done) {
+          reference3.delete().then(null, function (err) {
+              expect(err.message).toBe(integrityErrorMappedPureBinaryMessage, "invalid error message");
+              done();
+          }).catch(function(err) {
+              console.log(err);
+              done.fail();
+          });
+        });
+
+        it("if it's an integrity error and related table then from_name should be displayed", function (done) {
+          reference4.delete().then(null, function (err) {
+              expect(err.message).toBe(integrityErrorMappedFromnameMessage, "invalid error message");
+              done();
+          }).catch(function(err) {
+              console.log(err);
+              done.fail();
+          });
+        });
+        it("if it's an integrity error and no displayName was found then use table name passed by ermrest using mock", function (done) {
+          var mockUri = "/ermrest/catalog/"+catalog_id+"/entity/M:=" + schemaName + ":" + tableNameWithoutDisplayName;
+
+          nock(url, ops)
+              .delete(mockUri)
+              .reply(409, integrityErrorServerResponseArb)
+              .persist();
+          reference2.delete().then(null, function (err) {
+              expect(err.message).toBe(integrityErrorMappedSiteAdminMessage, "invalid error message");
+              done();
+          }).catch(function(err) {
+              console.log(err);
+              done.fail();
+          });
+
+        });
+
+        it("if it's an integrity error without deletion flag passed then we should allow the ermrest error to the client.", function (done) {
+           nock(url, ops)
+               .get("/ermrest/catalog/1234/schema")
+               .reply(409, integrityErrorServerResponse)
+               .persist();
+
+           server.catalogs.get("1234").then(null, function(err) {
+               expect(err.code).toBe(409, "invalid error code");
+               expect(err.message).toBe(integrityErrorWithoutDelMessage, "invalid error message");
```

Comment:
If this is going to the general (third) case, it should not have the prefix. But in your expected value the prefix is there.

Reply:
This is the case when there is no DELETE flag.  In this case everything that is coming from the server would be passed.